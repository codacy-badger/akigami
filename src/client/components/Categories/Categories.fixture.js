import Categories from './Categories';

export default {
  component: Categories,
  props: {
    items: [
      { id: 1, active: true, icon: 'account-multiple', title: 'Пользователи' },
      { id: 2, active: false, icon: 'book-variant', title: 'Закладки' },
      { id: 3, active: false, icon: 'cupcake', title: 'Маффины' },
    ],
    onSelect: item => console.log('Select', item),
  },
};
