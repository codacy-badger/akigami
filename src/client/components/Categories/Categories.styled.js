import { css } from 'emotion';
import styled from '@emotion/styled'
import { Button } from 'react-bootstrap';
import I from '../Icon';

export const List = styled('div')`
  display: flex;
  flex-direction: column;
`;

const active = css`
  border-color: #d54343 !important;
  box-shadow: none !important;
`;

const none = css`
  border-color: transparent;
`;

export const Item = styled(Button)`
  ${p => (p.active ? active : none)};
  display: flex;
  align-items: center;
  color: #666;
  border: 1px solid;
  background-color: white;


  &:not(:last-child) {
    margin-bottom: 8px;
  }

  &:hover {
    border-color: #ddd;
    color: inherit;
  }

  &:focus,
  &:active {
    box-shadow: none;
  }
`;

export const Icon = styled(I)`
  font-size: 20px;
`;

export const Title = styled('span')`
  margin-left: 12px;
  line-height: 28px;
`;
