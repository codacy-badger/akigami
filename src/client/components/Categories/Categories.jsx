import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Block from '../Block';
import { List, Item, Icon, Title } from './Categories.styled';

class Categories extends PureComponent {
    static defaultProps = {
        onSelect: () => {},
    }
    static propTypes = {
        items: PropTypes.array.isRequired,
        onSelect: PropTypes.func,
    }
    render() {
        const { items, onSelect } = this.props;
        return (
            <Block bordered shadow title="Фильтр">
                <List>
                    {items.map(item => (
                        <Item
                            key={item.id}
                            active={item.active}
                            onClick={() => onSelect(item)}
                        >
                            {item.icon && <Icon type={item.icon} />}
                            <Title>{item.title}</Title>
                        </Item>
                    ))}
                </List>
            </Block>
        );
    }
}

export default Categories;
