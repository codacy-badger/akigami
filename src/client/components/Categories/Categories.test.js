import { create as renderer } from 'react-test-renderer';
import createTestContext from 'react-cosmos-test/generic';
import Categories from './Categories.fixture';

const res = createTestContext({ renderer, fixture: Categories });
beforeEach(res.mount);
test('Categories snapshot', () => {
  expect(res.getWrapper().toJSON()).toMatchSnapshot();
});

