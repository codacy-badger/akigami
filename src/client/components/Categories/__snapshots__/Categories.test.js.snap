// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories snapshot 1`] = `
<div
  className="css-nczllq"
>
  <div
    className="css-czr64w"
  >
    <h3
      className="css-1r0b13k"
    >
      Фильтр
    </h3>
  </div>
  <div
    className="css-bqvltc"
  >
    <div
      className="css-1iyoj2o"
    >
      <button
        className="css-rpiht9 btn btn-default active"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="mdi mdi-account-multiple css-end85d"
        />
        <span
          className="css-1gjna6e"
        >
          Пользователи
        </span>
      </button>
      <button
        className="css-69fdvv btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="mdi mdi-book-variant css-end85d"
        />
        <span
          className="css-1gjna6e"
        >
          Закладки
        </span>
      </button>
      <button
        className="css-69fdvv btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="mdi mdi-cupcake css-end85d"
        />
        <span
          className="css-1gjna6e"
        >
          Маффины
        </span>
      </button>
    </div>
  </div>
</div>
`;
